### PR TITLE
frr: Add terminal and cliconf plugins

### PR DESCRIPTION
**Pull request by NilashishC**
##### SUMMARY
- Added terminal and cliconf plugins for Free Range Routing platform
##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
cliconf/frr.py
terminal/frr.py

### CODE REVIEW

File: lib/ansible/plugins/cliconf/frr.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            'supports_commit_comment': False,
+            'supports_multiline_delimiter': False,
+            'supports_diff_match': False,
+            'supports_diff_ignore_lines': False,
+            'supports_generate_diff': False,
+            'supports_replace': False
+        }
+
+    def get_capabilities(self):
+        result = dict()
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/frr.py]:
```suggestion
        result = super(Cliconf, self).get_capabilities()
```

File: lib/ansible/plugins/cliconf/frr.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        result['rpc'] = self.get_base_rpc() + ['get']
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/frr.py]: `'get'` is in base_rpc already?
```suggestion
```

File: lib/ansible/plugins/cliconf/frr.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        result['network_api'] = 'cliconf'
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/frr.py]:
```suggestion
```

File: lib/ansible/plugins/cliconf/frr.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        result['device_info'] = self.get_device_info()
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/frr.py]:
```suggestion
```

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Review by Qalthos — APPROVED** [review]

File: lib/ansible/plugins/cliconf/frr.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            'output': ['text']
+        }
+
+    def get_device_operations(self):
+        return {
+            'supports_diff_replace': False,
+            'supports_commit': False,
+            'supports_rollback': False,
+            'supports_defaults': False,
+            'supports_onbox_diff': False,
```
Comment (trishnaguha) [inline lib/ansible/plugins/cliconf/frr.py:86]: @NilashishC We can have `get_diff` method for frr which generates diff between running config and candidate config?

Comment (NilashishC) [reply lib/ansible/plugins/cliconf/frr.py:86]: Added get_diff @trishnaguha

**Comment by NilashishC** [comment]
@trishnaguha added get_diff and run_commands

**Review by ganeshrn — APPROVED** [review]

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**